Build Linux binaries for ARM64

getchoo opened the pull request:
https://github.blog/changelog/2025-01-16-linux-arm64-hosted-runners-now-available-for-free-in-public-repositories-public-preview/

Hopefully this can be released with 9.3, but we should let it sit for a bit and get tested before that

Some caveats:

- ~~AppImage builds are currently blocked by upstream ARM support in the [linuxdeploy AppImage plugin](https://github.com/linuxdeploy/linuxdeploy-plugin-appimage/issues/32). It'd be good to build against ARM in the meantime though :)~~

- Qt >= 6.7 is required (Binary arm64 builds are not available for prior versions)

- Ubuntu 24.04 is the minimum required target (this is due to the Qt binaries depending on glibc >= 2.37)

<!--
Hey there! Thanks for your contribution.

Please make sure that your commits are signed off first.
If you don't know how that works, check out our contribution guidelines: https://github.com/PrismLauncher/PrismLauncher/blob/develop/CONTRIBUTING.md#signing-your-work
If you already created your commits, you can run `git rebase --signoff develop` to retroactively sign-off all your commits and `git push --force` to override what you have pushed already.

Note that signing and signing-off are two different things!
-->

getchoo committed:
ci: build for arm on linux

Signed-off-by: seth <getchoo@tuta.io>

getchoo commented:
Draft until/blocked by https://github.com/PrismLauncher/PrismLauncher/pull/2895 (negotiable depending on if we need to drop the Qt version for Ubuntu 20.04) and https://github.com/jurplel/install-qt-action/issues/248

DioEgizio commented:
Can't we use ubuntu-22.04-arm for more compatibility? For the install-qt issue, can't we use the qt version from Ubuntu repos for now?

getchoo commented:
> For the install-qt issue, can't we use the qt version from Ubuntu repos for now?

We *could*, but the version in 22.04 is 6.2. Not really sure if it's worth dropping it to that

crueter commented:
> AppImage builds are currently blocked by upstream ARM support in the https://github.com/linuxdeploy/linuxdeploy-plugin-appimage/issues/32. It'd be good to build against ARM in the meantime though :)

This isn't actually required, linuxdeploy has bundled the appimage plugin into the main linuxdeploy AppImage for a while now, including on aarch64, so you can delete this line with no consequence.

```
wget "https://github.com/linuxdeploy/linuxdeploy-plugin-appimage/releases/download/continuous/linuxdeploy-plugin-appimage-${{ matrix.architecture }}.AppImage"
```

getchoo commented:
Thanks for noticing that, it's a huge help :)

getchoo commented:
install-qt-action support should land in https://github.com/jurplel/install-qt-action/pull/273. I'll undraft this after

getchoo commented:
These builds have been tested by our very cool Discord members. Only waiting on the above now

getchoo committed:
ci(package/linux): use dpkg to determine file paths and variables

Should hopefully make things less brittle across different architectures

Signed-off-by: Seth Flynn <getchoo@tuta.io>

getchoo committed:
ci(setup-deps): always use sccache, simplify restore key

sccache is available on arm runners. we can use the restore key for an
easy, unique restore key in the cache too (it also prevents us from
re-using the ccache caches!)

Signed-off-by: Seth Flynn <getchoo@tuta.io>

getchoo committed:
cmake: enforce explicit artifact name

It's much easier to determine this in CI and ensure our artifact names
are correct (I have made some accidents). They (and thus the updater)
can also easily be left out of local builds -- and probably should've
always been

Signed-off-by: Seth Flynn <getchoo@tuta.io>

getchoo committed:
cmake: use build platform from environment

This allows all CI builds to be deemed "official"

Signed-off-by: Seth Flynn <getchoo@tuta.io>
